jdk18 failing to compile MethodHandleResolver.linkDynamicConstant linkCallSite

https://openj9-jenkins.osuosl.org/job/Pipeline-OpenJDK-Acceptance/133/
```
00:03:32  /home/jenkins/workspace/Build_JDKnext_aarch64_linux_OpenJDK/build/linux-aarch64-server-release/support/j9jcl/java.base/share/classes/java/lang/invoke/MethodHandleResolver.java:135: error: method linkDynamicConstant in class MethodHandleNatives cannot be applied to given types;
00:03:32  			result = MethodHandleNatives.linkDynamicConstant(classObject, 0, bsm, name, typeClass, (Object)staticArgs);
00:03:32  			                            ^
00:03:32    required: Object,Object,Object,Object,Object
00:03:32    found:    Class<CAP#1>,int,MethodHandle,String,Class<CAP#2>,Object
00:03:32    reason: actual and formal argument lists differ in length
00:03:32    where CAP#1,CAP#2 are fresh type-variables:
00:03:32      CAP#1 extends Object from capture of ?
00:03:32      CAP#2 extends Object from capture of ?
00:03:32  /home/jenkins/workspace/Build_JDKnext_aarch64_linux_OpenJDK/build/linux-aarch64-server-release/support/j9jcl/java.base/share/classes/java/lang/invoke/MethodHandleResolver.java:194: error: method linkCallSite in class MethodHandleNatives cannot be applied to given types;
00:03:32  		result[0] = MethodHandleNatives.linkCallSite(classObject, 0, bsm, name, type, (Object)staticArgs, appendixResult);
00:03:32  		                               ^
00:03:32    required: Object,Object,Object,Object,Object,Object[]
00:03:32    found:    Class<CAP#1>,int,MethodHandle,String,MethodType,Object,Object[]
00:03:32    reason: actual and formal argument lists differ in length
00:03:32    where CAP#1 is a fresh type-variable:
00:03:32      CAP#1 extends Object from capture of ?
```
@tajila fyi

--- a/jcl/src/java.base/share/classes/java/lang/invoke/MethodHandleResolver.java
+++ b/jcl/src/java.base/share/classes/java/lang/invoke/MethodHandleResolver.java
@@ -151,7 +151,11 @@ final class MethodHandleResolver {
 		 */
 		try {
 /*[IF OPENJDK_METHODHANDLES]*/
-			result = MethodHandleNatives.linkDynamicConstant(classObject, 0, bsm, name, typeClass, (Object)staticArgs);
+			result = MethodHandleNatives.linkDynamicConstant(classObject,
+					/*[IF JAVA_SPEC_VERSION < 18] The second formal parameter was removed in Java 18 by 8272614. */
+					0,
+					/*[ENDIF] JAVA_SPEC_VERSION < 18 */
+					bsm, name, typeClass, (Object)staticArgs);
 /*[ELSE] OPENJDK_METHODHANDLES*/
 			result = (Object)invokeBsm(bsm, staticArgs);
 /*[ENDIF] OPENJDK_METHODHANDLES*/
@@ -256,7 +260,11 @@ final class MethodHandleResolver {
 		 * TODO: Investigate if the Throwable can be wrapped in a MemberName. This will help prevent type checks in the
 		 * interpreter since result[0] will always be a MemberName. This will improve performance.
 		 */
-		result[0] = MethodHandleNatives.linkCallSite(classObject, 0, bsm, name, type, (Object)staticArgs, appendixResult);
+		result[0] = MethodHandleNatives.linkCallSite(classObject,
+				/*[IF JAVA_SPEC_VERSION < 18] The second formal parameter was removed in Java 18 by 8272614. */
+				0,
+				/*[ENDIF] JAVA_SPEC_VERSION < 18 */
+				bsm, name, type, (Object)staticArgs, appendixResult);
 
 		/* result[1] stores a MethodHandle object, which is used as the last argument to the caller method. */
 		result[1] = appendixResult[0];

